### ISSUE TITLE
Issue with ILM rollover step when user adds lifecycle settings to index

### ISSUE THREAD

**Issue by bmcconaghy** [elastic/elasticsearch#35211]
When a user adds a lifecycle policy with rollover to an index through the settings API (or at index creation time) and does not do so through a template, the index created by the rollover action does not have index lifecycle management settings.  This will be surprising to users as many will assume that the settings will be carried forward across the rollover.

A few possible solutions have been proposed (all with downsides):
- We just say this can’t/won’t apply to existing indices period and the only way the lifecycle settings ever wind up on an index is when the index is created with a template. (issues: forces folks to use a template when maybe they don't want to, can't add ilm to existing indices)
- Look at the resulting index and if it does not have lifecycle settings, copy them over from the index that was rolled over. (issues: if the index is created from a template and the template does not have lifecycle settings, lifecycle settings will be applied when the user does not want them)
- Allow for settings to be applied on rollover to be defined in the rollover step of lifecycle policy (issues: template could have lifecycle policy settings and those will be overridden)

**Comment by talevy** [comment]
Do move this forward, should we be copying over the `index.lifecycle.name` and `index.lifecycle.rollover_alias` to the new index?

**Comment by colings86** [comment]
We agreed in one of the syncs a few weeks ago that we are not going to address this issue for now as all of the solutions proposed have significant downsides and would be hard for both the user and us to reason about.

Today we are able to describe what each element (policy, index template, and bootstrap index) does and what part it plays with ILM so we have added a note to the documentation to point out that if you are using rollover you will need to also use an index template so the ILM settings are applied to new indexes created through the rollover API.

We may want to address this in a future release but we will need to think of a better solution than the ones we have already discussed.
